Приложение для терминала. Часть 2

Dimaggio-IT opened the pull request:
---
:mortar_board: [Приложение для терминала. Часть 2](https://up.htmlacademy.ru/nodejs-api/6/user/2256781/tasks/6)

Dimaggio-IT committed:
Доработка дз Приложение для терминала. Часть 1

добавил динамический импорт и регистрацию
команд
- установил пакет glob
- изменил main.cli.ts
- внёс правки по совету наставника

Dimaggio-IT committed:
продолжение работы над терм. приложением

добавил тип MockServerData
добавил интерфейс OfferGenerator
добавит TSVOfferGenerator
добавил команду формирования объявлений
добавил пакет got.
добавил запрос для получения данных
добавил генератор в команду GenerateCommand

ElkostVN commented on src/shared/helpers/common.ts:
```diff
@@ -0,0 +1,21 @@
+}
+
+export function getRandomItems<T>(items: T[], length?: number): T[] {
+  let startPosition: number;
+  let endPosition: number;
+
+  if (length) {
+    startPosition = 0;
+    endPosition = length;
+  } else {
+    startPosition = generateRandomValue(0, items.length - 1);
+    endPosition = startPosition + generateRandomValue(startPosition, items.length);
+  }
+  return items.slice(startPosition, endPosition);
```
const startPosition = length ? 0 : generateRandomValue(0, items.length - 1);
  const endPosition = length ?? (startPosition + generateRandomValue(startPosition, items.length));

  return items.slice(startPosition, endPosition);

Dimaggio-IT replied:
ок

ElkostVN commented on src/shared/libs/offer-generator/tsv-offer-generator.ts:
```diff
@@ -0,0 +1,91 @@
+const MIN_REVIEW_COUNT = 0;
+const MAX_REVIEW_COUNT = 4;
+
+
```
-

Dimaggio-IT replied:
ок
+ сделал константный объект куда добавил нужные для генерации константы

Dimaggio-IT committed:
продолжение работы над терм. приложением. Часть 3

рефакторинг
-  TSVFileReader стал экстендить EventEmitter
- getRandomItems стал "красивее"
- чуть изменил мок-сервис (mock-server-data.type.ts)

добавлена ф-ия createOffer. Создаёт из прочитанной строки файла
объект TOffer и заполняет его

заиспользовал streams
- добавлен класс TSVFileWriter реализующая запись в файл
чанками
- в класс TSVFileReader добавлен метод read реализующий чтение из файла
чанками

Dimaggio-IT committed:
продолжение работы над терм. приложением. Часть 4

произвел рефакторинг классов команд
- изменил интерфейс (теперь используется геттер для получения имени команды)
- внёс изменения в классы команд, с учётом использования геттера в интерфейсе

Dimaggio-IT committed:
добавил игнор для .env
